packages/amplify-cli/src/extensions/amplify-helpers/push-resources.ts

```diff
@@ -87,7 +87,28 @@ export const pushResources = async (
 
   // Verify any environment parameters before push operation
   const envParamManager = (await ensureEnvParamManager()).instance;
-  envParamManager.verifyExpectedEnvParameters();
+
+  const promptMissingParameter = async (parameter: string, envParamManager: IEnvironmentParameterManager): Promise<void> => {
+    printer.warn(`Could not find value for parameter ${parameter}`);
+    const [, categoryName, resourceName, parameterName] = parameter.split('_');
```

Comment:
This kind of magic destructuring is happening in enough places in this change that I think it's worth it to factor it out into a helper method. It's not explicit why the first value is being ignored (presumably because it equals `AMPLIFY`?). It's also not explicit why in `packages/amplify-environment-parameters/src/environment-parameter-manager.ts` line 201, the final value is ignored.

Factoring this destructuring out to another function that returns a predictable value is probably only half of a solution. If we are implicitly expecting our parameter names to match this naming structure, we should probably have a type that explicitly defines that structure in the first place, rather than implicitly capturing it in magic strings.